### PR TITLE
Feature: make frontend timezone un-aware

### LINKED ISSUE #947: [BUG] How to handle TZ differences on frontend.
### Description

Rather than continuing to hijack the thread in which it came up (https://github.com/paperless-ngx/paperless-ngx/issues/926#issuecomment-1125421989) I wanted to discuss this to see what everyone thinks is the best solution. Sorry its long, dates are... complicated AF.

To summarize: the backend currently passes UTC timestamps for document dates. There is no "timezone" but UTC dates always have an offset (e.g. +1:00) that is related to timezone ([they are not the same](https://spin.atomicobject.com/2016/07/06/time-zones-offsets/)) but the backend has an explicit `PAPERLESS_TIME_ZONE` which can be set and then can differ from the frontend which doesnt get that variable but rather pulls its timezone from the system (Javascript does this). That can lead to situations like the above linked comment where the date looks wrong because its being interpolated to a different timezone than the backend is set to.

First of all, **I think this is likely a rare situation**. I doubt most users would intentionally do this but perhaps thats more of a reason to discuss this, i.e. to warn them. Also it would happen if the user is traveling etc.

Proposed solutions:

Edit: #957!

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

This PR tries to address the fact that the use of time-aware dates for the user-editable `document.created` can cause unexpected date behavior on the frontend. See the linked issue.

Since the frontend UI has no way to set a time, it should really not be handling time at all. But the backend is TZ aware, so unless we want to change the `created` field from DateTime to Date (presumably a breaking change for the API), this is an attempt to workaround. If everyone feels that is the better course I would be happy with that too.

A little more explanation of this:

- Adds `created_date` field to the API (not the db) which is just the localized date. In other words `2012-12-12 00:00:00+01:00` returns `2012-12-12` and then the frontend can treat that as an absolute date.
- The document endpoint `update` (PUT) accepts both `created_date` and still `created` (so not a breaking change). If both are sent it defers to `created`.
- When `created_date` is passed in (without a time ofc) `created` is set to midnight that date in the timezone the backend is set to use.

## TODO:

- [x] Update docs (api.rst)

Fixes #947 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by stumpylog** [comment]
This seems alright to me as a fix.  Since the backend knows about the timezone, let it handle sending localized dates.

File: src/documents/serialisers.py [hunk 1]
```diff
@@ -216,6 +220,18 @@ def get_archived_file_name(self, obj):
         else:
             return None
 
+    def update(self, instance, validated_data):
+        if "created_date" in validated_data and "created" not in validated_data:
+            new_datetime = datetime.datetime.combine(
+                validated_data.get("created_date"),
+                datetime.time(0, 0, 0, 0, tz.gettz(settings.TIME_ZONE)),
```
Comment (stumpylog) [inline src/documents/serialisers.py:227]: I think zoneinfo is the preferred "modern" method for getting the tzinfo

Comment (shamoon) [reply src/documents/serialisers.py:227]: Totally defer on this, obviously. I got this from https://github.com/paperless-ngx/paperless-ngx/blob/9a1bd9637cbef0e7318a3c61cae14805c6510cea/src/documents/tests/test_date_parsing.py but happy to change if thats better

Comment (shamoon) [reply src/documents/serialisers.py]: Cool yea did some reading, thanks

**Pull request merged**